torch/fx/passes/tensorify_python_scalars.py

```diff
@@ -0,0 +1,241 @@
+            if isinstance(expr, BooleanAtom):
+                dtype = torch.bool
+                c = bool(expr)
+            elif isinstance(expr, sympy.Integer):
+                dtype = torch.int64
+                c = int(expr)
+            elif isinstance(expr, sympy.Number):
+                dtype = torch.float64
+                c = float(expr)
+
+            node = graph.call_function(
+                torch.ops.aten.scalar_tensor.default, (c,), {"dtype": dtype}
+            )
+            node.meta["val"] = torch.tensor(c, dtype=dtype)
+            expr_to_tensor_proxy[expr] = fx.Proxy(
+                node,
+                tracer=tracer,
+            )
+
+        if expr in expr_to_tensor_proxy:
+            return expr_to_tensor_proxy[expr]
+
+        # don't cache
+        if isinstance(expr, Symbol):
+            return sympy_interp(Analysis, expr_to_tensor_proxy, expr)  # type: ignore[arg-type]
+
+        # hash cons on arguments, run expr handler
+        expr_to_tensor_proxy[expr] = _run_sympy_handler(
+            Analysis,
+            [_sympy_interp(arg) for arg in expr.args],  # type: ignore[arg-type]
+            expr,
+        )
+
+        return expr_to_tensor_proxy[expr]
+
+    nodes = list(graph.nodes)
+    for i, node in enumerate(nodes[:-1]):
+        with graph.inserting_before(
+            nodes[i + 1] if node not in placeholders else first_non_placeholder
+        ):
+            # Look for tensor.item() calls on placeholders
+            if unbacked_bindings := node.meta.get("unbacked_bindings"):
+                for s in unbacked_bindings.keys():
+                    if (
+                        node is not None
+                        and node.op == "call_function"
+                        and node.target is torch.ops.aten._local_scalar_dense.default
+                    ):
+                        dtype = node.args[0].meta["val"].dtype
+                        if dtype != torch.float64:
+                            continue
+
+                        assert isinstance(node.args[0], fx.Node), node.args[0]
+
+                        expr_to_tensor_proxy[s] = fx.Proxy(node.args[0], tracer=tracer)
+                        expr_to_sym_proxy[s] = fx.Proxy(node, tracer=tracer)
+
+            elif (sym_expr := _get_sym_val(node)) is not None:
+                if sym_expr not in expr_to_sym_proxy and not isinstance(
+                    sym_expr, (sympy.Number, sympy.logic.boolalg.BooleanAtom)
+                ):
+                    expr_to_sym_proxy[sym_expr] = fx.Proxy(node, tracer=tracer)
+
+            # Look for functions to convert
+            if node.op == "call_function" and node.target is torch.ops.aten.mul.Tensor:
+                args = []
+                transform = False
+                compute_dtype, result_dtype = elementwise_dtypes(
+                    node.meta["val"],
+                    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+                )
```

Comment:
Because you're only consulting the result dtype, don't bother with `elementwise_dtypes`, just directly call `get_computation_dtype` on the node.meta["val"].dtype